sound/soc/sof/sof-audio.h

```diff
@@ -420,6 +424,15 @@ struct snd_sof_widget {
 	 */
 	bool payload_with_output_fmt;
 
+	/*
+	 * There are two styles of SOF module instance initialization blob:
+	 *   - base module config + module specific extension
+	 *   - base module config + base module config extension
+	 *
+	 * This flag is used to tell which initialization blob style a widget is using.
+	 */
+	int init_blob_style;
```

Comment:
this is IPC4 specific isnt it? Does it need to go in struct snd_sof_widget? Can we find a better place?

Reply:
snd_sof_widget is shared between IPC3 and IPC4, I don't think we have a IPC4 specific struct which is a good fit for this.

we can later add snd_sof_ipc3_widget/snd_sof_ipc4_widget if necessary.

Reply:
How about  struct sof_ipc4_process? We know that the base_fw modules already use module specific extension.

Reply:
payload_with_output_fmt is already IPC4-specific @ranj063, looks like another poorly defined partitioning.

Reply:
> payload_with_output_fmt is already IPC4-specific @ranj063, looks like another poorly defined partitioning.

Maybe fix both of them?